content/general/concepts/software-development-life-cycle/software-development-life-cycle.md

```diff
@@ -0,0 +1,39 @@
+Subjects:
+  - 'Mobile Development'
+  - 'Web Development'
+Tags:
+  - 'Development'
+  - 'Models'
+CatalogContent:
+  - 'learn-how-to-code'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+The **Software Development Life Cycle (SDLC)** is a process that is used to efficiently produce cost-effective and high-quality software through a series of structured steps or phases. Essentially, it is a standard industry approach to software development that outlines an entire project and timeline.
+
+## Phases
+
+The SDLC phases are as follows:
+
+1. Planning is the phase where requirement gathering and analysis takes place. It involves brainstorming ideas to solve some problem as a result of customer feedback or need.
+2. Defining or requirement defining is the phase that involves creating a Software Requirement Specification (SRS). The SRS document defines the purpose, product, and technical requirements. This allows everyone working on the software to understand the product's mission.
+3. Designing is based on the SRS and results in a Design Document Specification (DDS). The DDS is reviewed by all stakeholders based on a set of criteria.
+4. Development or building deals with reviewing and implementing code that will be integrated into the software.
+5. Testing and integration is the phase that [software tests](https://www.codecademy.com/resources/docs/general/software-testing) are performed to determine if the software satisfies the SRS. If it meets requirements, then the product or feature is integrated into the software.
+6. Maintaining is an ongoing phase that involves supporting and updating the software as needed.
+
+## Models
+
+The Waterfall Method was the first model to be created, however, many models now exist as part of the SDLC:
+
+- Agile: Iterative development and continuous integration. The scrum methodology is a common implementation of the agile model in which teams work in sprints. Sprints are time-constrained periods where a team works to complete a certain goal.
+- Big Bang: Named after the physical theory, this model involves creating a product through available resources for technical development. There is little planning or documenting.
+- Iterative: Depending on the requirements of the product, the software is modified upon each iteration and changes are released as a new version of the software.
+- Prototype: A prototype, that does not necessarily represent the final logic of the software, is created and tested until it meets requirements for deployment or integration.
+- Rapid Application Development (RAD): Meets requirements through prototyping iteratively.
+- Spiral: Based on the waterfall and iterative models.
+- V: Based on the waterfall model.
+- Waterfall: The goal is to create a minimum viable product (MVP). Each phase in the SDLC must be completed before moving on to the next phase.
```

Comment:
I can see terms made for each of these models. Should I create issues on a few of these?

Reply:
That sounds like a good plan.